internal/cmd/storagebox/enable_protection.go

```diff
@@ -0,0 +1,94 @@
+package storagebox
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"github.com/hetznercloud/cli/internal/cmd/base"
+	"github.com/hetznercloud/cli/internal/cmd/cmpl"
+	"github.com/hetznercloud/cli/internal/cmd/util"
+	"github.com/hetznercloud/cli/internal/hcapi2"
+	"github.com/hetznercloud/cli/internal/state"
+	"github.com/hetznercloud/hcloud-go/v2/hcloud"
+)
+
+func getChangeProtectionOpts(enable bool, flags []string) (hcloud.StorageBoxChangeProtectionOpts, error) {
+	opts := hcloud.StorageBoxChangeProtectionOpts{}
+
+	var unknown []string
+	for _, arg := range flags {
+		switch strings.ToLower(arg) {
+		case "delete":
+			opts.Delete = enable
+		default:
+			unknown = append(unknown, arg)
+		}
+	}
+	if len(unknown) > 0 {
+		return opts, fmt.Errorf("unknown protection level: %s", strings.Join(unknown, ", "))
+	}
+
+	return opts, nil
+}
+
+func changeProtection(s state.State, cmd *cobra.Command,
+	storageBox *hcloud.StorageBox, enable bool, opts hcloud.StorageBoxChangeProtectionOpts) error {
+
+	/*
+		// TODO this needs to be fixed in the hcloud-go library first
+		if opts.Delete == nil {
+			return nil
+		}
```

Comment:
This needs to be changed in the API first to make the `delete` field optional. Talks are ongoing.